service/http.directory/src/main/java/org/kaazing/gateway/service/http/directory/HttpDirectoryServiceHandler.java

```diff
@@ -104,6 +104,10 @@ void setPatterns(List<PatternCacheControl> patterns) {
         this.patterns = patterns;
     }
 
+    void emptyUrlCacheControlMap() {
+        urlCacheControlMap.clear();
+    }
```

Comment:
The urlCacheControlMap is static so that it caches the resolved header for each url. If it weren't static, this would have to be resolved for every url, at each request.